onadata/libs/utils/logger_tools.py

```diff
@@ -213,6 +218,32 @@ def dict2xform(jsform, form_id):
     return xml_head + dict2xml(jsform) + xml_tail
 
 
+@contextlib.contextmanager
+def get_instance_lock(xml_hash: str) -> bool:
+    int_lock = int(xml_hash, 16) & 0xfffffffffffffff
+    is_postgresql = connection.vendor == 'postgresql'
+    acquired = False
+
+    with transaction.atomic():
+        try:
+            if is_postgresql:
+                cur = connection.cursor()
+                cur.execute('SELECT pg_try_advisory_lock(%s::bigint);', (int_lock,))
+                acquired = cur.fetchone()[0]
+            else:
+                prefix = os.getenv('KOBOCAT_REDIS_LOCK_PREFIX', 'kc-lock')
+                key_ = f'{prefix}:{int_lock}'
+                redis_lock = settings.REDIS_LOCK_CLIENT.lock(key_, timeout=60)
+                acquired = redis_lock.acquire(blocking=False)
+            yield acquired
```

Comment:
I don't think support for something other than PostgreSQL is needed. We already depend on Postgres for many things.

Moot point, then, but just to say it: I think all `os.getenv()` for configuration (and default values) should be in the settings files